search/Binary_Search.c

```diff
@@ -21,7 +21,7 @@ int main() {
  top = n_terms;
  
  do {
-  mid = (bottom + top) / 2;
+  mid = bottom +  (top - bottom)/2 ; // same as (bottom + top / 2) but considers overflow condition when bottom + top might be larger than int 
```

Comment:
That's a good optimization.
Although it may overflow in some cases. But that's fine anyway.
👍